quickjs.c

```diff
@@ -1309,11 +1285,35 @@ static JSValue js_int32(int32_t v)
     return JS_MKVAL(JS_TAG_INT, v);
 }
 
+static JSValue js_float64(double d)
+{
+    return JS_NewFloat64__(d);
+}
+
 static JSValue js_uint32(uint32_t v)
 {
     if (v <= INT32_MAX)
         return js_int32(v);
-    return js_float64(v);
+    else
+        return js_float64(v);
+}
+
+static JSValue js_int64(int64_t v)
+{
+    if (v == (int32_t)v)
+        return js_int32(v);
+    else
+        return js_float64(v);
+}
+
+#define JS_NewInt64(ctx, val)  js_int64(val)
```

Comment:
Why is this necessary, if the function is already declared in the header file?

Reply:
The external APIs should not be called from the implementation to avoid potential overhead.  I did not change all calls to `JS_NewInt64` in the code, the macro takes care of it.  I should define equivaoent macros for other `JS_Newxxx` with an explanatory comment.  This is work in progress.